tools/dependency/release_dates.py

```diff
@@ -60,11 +61,33 @@ def format_utc_date(date):
     return date.date().isoformat()
 
 
+def get_latest_release(repo, version_min):
+    latest_release = repo.get_latest_release()
+    # In some scenarios we have seen that the get_latest_release is fetching the latest release by date
+    # and not really the latest release in chronological order. Add a backup logic to guard against such cases.
+    latest_release_version = parse_version(latest_release.tag_name)
+    current_version = parse_version(version_min)
+    if (latest_release_version >= current_version):
```

Comment:
im wondering about this - i think i can imagine a situation where the latest_release's version is higher than the current version but is still not the highest numbered version

Reply:
Could you please give an example for this case?

Reply:
eg if we have version `1.16.3` and the last release (by date) is `1.17.2` but the highest versioned release is `1.18.0` it would return `1.17.2`

Reply:
Removed this logic now, and have only the iteration over all releases. Please take a look.